src/Bicep.LangServer/Deploy/DeploymentHelper.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Resources.Models;
+using Bicep.Core;
+
+namespace Bicep.LanguageServer.Deploy
+{
+    public class DeploymentHelper
+    {
+        public static async Task<string> CreateDeployment(
+            IDeploymentCollectionProvider deploymentCollectionProvider,
+            ArmClient armClient,
+            string template,
+            string parameterFilePath,
+            string id,
+            string scope,
+            string location)
+        {
+            if ((scope == LanguageConstants.TargetScopeTypeSubscription ||
+                scope == LanguageConstants.TargetScopeTypeManagementGroup) &&
+                string.IsNullOrWhiteSpace(location))
+            {
+                return LangServerResources.MissingLocationDeploymentFailedMessage;
+            }
+
+            DeploymentCollection? deploymentCollection;
+            var resourceIdentifier = new ResourceIdentifier(id);
+
+            if (scope == LanguageConstants.TargetScopeTypeResourceGroup ||
+                scope == LanguageConstants.TargetScopeTypeSubscription ||
+                scope == LanguageConstants.TargetScopeTypeManagementGroup)
+            {
+                try
+                {
+                    deploymentCollection = deploymentCollectionProvider.GetDeploymentCollection(armClient, resourceIdentifier, scope);
+                }
+                catch (Exception e)
+                {
+                    return string.Format(LangServerResources.DeploymentFailedWithExceptionMessage, e.Message);
+                }
+            }
+            else
+            {
+                return LangServerResources.UnsupportedScopeDeploymentFailedMessage;
+            }
+
+            if (deploymentCollection is not null)
+            {
+                JsonElement parameters;
+
+                if (string.IsNullOrWhiteSpace(parameterFilePath))
+                {
+                    parameters = JsonDocument.Parse("{}").RootElement;
+                }
+                else
+                {
+                    try
+                    {
+                        string text = File.ReadAllText(parameterFilePath);
+                        parameters = JsonDocument.Parse(text).RootElement;
+                    }
+                    catch(Exception e)
+                    {
+                        return string.Format(LangServerResources.InvalidParameterFileDeploymentFailedMessage, e.Message);
+                    }
+                }
+
+                var deploymentProperties = new DeploymentProperties(DeploymentMode.Incremental)
```

Comment:
Should we add support for `DeploymentMode.Complete`?

Reply:
@shenglol , looking at the definition here - https://github.com/Azure/azure-sdk-for-net/blob/254abd0198311a3716d4d74d94935e4dcae95a94/sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/Models/DeploymentProperties.cs#L32, incremental sounds safe. Is there a reason why you think we should support Complete?